src/sysl/importers/import_swagger.py

```diff
@@ -94,200 +76,285 @@ def javaParam(match):
     m: i for (i, m) in enumerate('get put post delete patch parameters'.split())}
 
 
-def parse_typespec(tspec):
-
-    typ = tspec.get('type')
-
-    # skip invalid arrays
-    if not typ and 'items' in tspec:
-        del tspec['items']
-
-    descr = tspec.pop('description', None)
-
-    if typ == 'array':
-        assert not (set(tspec.keys()) - {'type', 'items'}), tspec
-
-        # skip invalid type
-        if '$ref' in tspec['items'] and 'type' in tspec['items']:
-            del tspec['items']['type']
-
-        (itype, idescr) = parse_typespec(tspec['items'])
-        assert idescr is None
-        return ('set of ' + itype, descr)
-
-    def r(t):
-        return (t, descr)
-
-    fmt = tspec.get('format')
-    ref = tspec.get('$ref')
-
-    if ref:
-        assert not set(tspec.keys()) - {'$ref'}, tspec
-        return r(ref.lstrip('#/definitions/'))
-    if (typ, fmt) == ('string', None):
-        return r('string')
-    if (typ, fmt) == ('boolean', None):
-        return r('bool')
-    elif (typ, fmt) == ('string', 'date'):
-        return r('date')
-    elif (typ, fmt) == ('string', 'date-time'):
-        return r('datetime')
-    elif (typ, fmt) == ('integer', None):
-        return r('int')
-    elif (typ, fmt) == ('integer', 'int32'):
-        return r('int32')
-    elif (typ, fmt) == ('integer', 'int64'):
-        return r('int64')
-    elif (typ, fmt) == ('number', 'double'):
-        return r('float')
-    elif (typ, fmt) == ('number', 'float'):
-        return r('float')
-    elif (typ, fmt) == ('number', None):
-        return r('float')
-    elif (typ, fmt) == ('object', None):
-        descrs = {
-            k: descr
-            for (k, v) in tspec['properties'].iteritems()
-            for descr in [parse_typespec(v)[1]]
-            if descr
-        }
-        assert not descrs, descrs
-        fields = ('{} <: {}'.format(k, parse_typespec(v)[0])
-                  for (k, v) in sorted(tspec['properties'].iteritems()))
-        return r('{' + ', '.join(fields) + '}')
-    else:
-        return r(str(tspec))
-
-
-def main():
-    [swagger_path, appname, package, outfile] = sys.argv[1:]
+def parse_args(argv):
+    p = argparse.ArgumentParser(description='Converts Swagger (aka Open API Specification) documents to a Sysl spec')
+    p.add_argument('swagger_path', help='path of input swagger document')
+    p.add_argument('appname', help='appname')
+    p.add_argument('package', help='package')
+    p.add_argument('outfile', help='path of output file')
+    return p.parse_args(args=argv[1:])
 
-    swag = yaml.load(open(swagger_path))
 
-    w = writer.Writer('sysl')
+def make_default_logger():
+    return logging.getLogger('import_swagger').setLevel(logging.WARN)
```

Comment:
This is a defect.  i started getting into the mindset of "how can i rewrite the code to reduce the number of  lines in the parts of the code that dont have unit test coverage, to appease the codecov bot"  so i collapsed this logger factory into a one liner. but `setLevel` returns `None`, not a logger.

so (a) now this doesnt work ;
and (b) this is an amusing example of how refactoring something without test coverage to game some test coverage metric can reduce code quality

Reply:
i've edited the offending commits and also extended test coverage to `make_default_logger`.